### PR TITLE
feat(doc) add some installation notes

### PR DESCRIPTION
**Pull request by bitsf**
for #4
add some installation notes for KIND k8s.

please help review how to organize such notes.

### CODE REVIEW

**Comment by bitsf** [comment]
image

hi @holyhope my PR check always failed, do you know the reason

**Comment by holyhope** [comment]
Can you rebase on this branch: https://github.com/goharbor/harbor-operator/tree/doc_installation_notes please?
And tell me if it works now

File: docs/installation.md [hunk 1]
```diff
@@ -74,3 +74,63 @@ Kubernetes API running (see [Supported platforms](https://github.com/goharbor/ha
...
+2. install nginx-ingress with NodePort
+
+   ```bash
+   helm install nginx stable/nginx-ingress --set-string controller.config.proxy-body-size=0 --set controller.service.type=NodePort
+   ```
+
+3. patch nginx-ingress to use special node
+
+   ```bash
+   kubectl patch deployments nginx-nginx-ingress-controller -p '{"spec":{"template":{"spec":{"containers":[{"name":"nginx-ingress-controller","ports":[{"containerPort":80,"hostPort":80},{"containerPort":443,"hostPort":443}]}],"nodeSelector":{"ingress-ready":"true"},"tolerations":[{"key":"node-role.kubernetes.io/master","operator":"Equal","effect":"NoSchedule"}]}}}}'
```
[the remark this answers is not in the document]
Comment (bitsf) [reply docs/installation.md]: hi @holyhope , I'm not quite familiar with how to write this patch, this code is just copied from KIND k8s guild.
Actually I think this is not even a good method here.
I think It's better if we can use one command to implement this logic when run helm install stable/nginx-ingress.
I think other params such as tolerations should not be modified here.
Can you help enhance the script ?

File: docs/installation.md [hunk 2]
```diff
@@ -74,3 +74,63 @@ Kubernetes API running (see [Supported platforms](https://github.com/goharbor/ha
...
+1. get the cert name
+
+   ```bash
+   kubectl get h harbor-sample -o jsonpath='{.spec.tlsSecretName}'
+   ```
+
+2. install cert for docker
+
+   ```bash
+   kubectl get secret $(kubectl get h harbor-sample -o jsonpath='{.spec.tlsSecretName}') -o jsonpath='{.data.ca\.crt}' | base64 --decode |sudo tee /etc/docker/certs.d/$LBAAS_DOMAIN/ca.crt
```
[the remark this answers is not in the document]
Comment (bitsf) [reply docs/installation.md]: this looks better

File: docs/installation.md [hunk 3]
```diff
@@ -74,3 +74,63 @@ Kubernetes API running (see [Supported platforms](https://github.com/goharbor/ha
...
+         authorization-mode: "AlwaysAllow"
+   extraPortMappings:
+   - containerPort: 80
+      hostPort: 80
+      protocol: TCP
+   - containerPort: 443
+      hostPort: 443
+      protocol: TCP
+   - role: worker
+   - role: worker
```
[the remark this answers is not in the document]
Comment (bitsf) [reply docs/installation.md]: because in my test, install redis cluster will fail if less then 3 nodes.

**Comment by bitsf** [comment]
> Can you rebase on this branch: https://github.com/goharbor/harbor-operator/tree/doc_installation_notes please?
> And tell me if it works now

It seems the lint check are all passed when you push. Did you do any special ?

**Comment by bitsf** [comment]
> > Can you rebase on this branch: https://github.com/goharbor/harbor-operator/tree/doc_installation_notes please?
> > And tell me if it works now
> 
> It seems the lint check are all passed when you push. Did you do any special ?

oh, I see you use another action brpaz/hadolint-action@master

**Pull request merged**